### ISSUE TITLE
Short Range High Accuracy Mode Functionality 

### ISSUE THREAD

**Issue by vlomba0** [sparkfun/SparkFun_Qwiic_TMF882X_Arduino_Library#6]
There doesn't seem to be any short-range mode functionality yet it'd be cool to have.

**Comment by vlomba0** [comment]
I can make a PR with the feature. I know the innards of this library were downloaded from the AMS website though so if they update the library it'll be a little harder to merge into this repo but yeah.

**Comment by PaulZC** [comment]
Hi Victor (@Bag132 ),

Sure, a Pull Request would be great - thank you.

Best wishes,
Paul

**Comment by vlomba0** [comment]
I have setRange and getRange functions working. 
However, the image in tof_bin_image.c is outdated in this repository. To get the functions working I had to update the library with the new firmware version I found at [https://ams.com/en/tmf8821#tab/tools](url) under the TMF882x_Driver_SDK_Source_v1.19.zip download. 
Should I include the new firmware image in my PR, or should someone else merge it in at another time? @PaulZC

**Comment by PaulZC** [comment]
Hi Victor,
Please include the updated library in the PR. We can diff it to make sure you haven't changed anything! ;-)
Best,
Paul

**Comment by vlomba0** [comment]
PR is posted

**Comment by PaulZC** [comment]
Thank you Victor,

Have you finished making changes?

Best,
Paul

**Comment by vlomba0** [comment]
Yeah, at first it was going to be controlled by using an `enum` but since the TMF882X SDK short-range functions use `bool` that meant all returns from the functions were going to be binary and you wouldn't be able to interpret a 0x00 from ACTIVE_RANGE vs an I2C error solely based on the return from `tmf882x_mode_app_is_shortrange_mode`. But yeah I'm done with it.

**Comment by PaulZC** [comment]
OK - thanks!
It may take us a little while to merge and test this, but we will get to it.
Thanks again,
Paul

**Comment by vlomba0** [comment]
Yeah happy to contribute.
Also, the entire underlying TMF882X SDK has been updated. I had an original version of the functions but I later found out that ams had already added `tmf882x_mode_app_is_shortrange_mode` and `tmf882x_mode_app_set_shortrange_mode` so I just copied them and pasted over mine.

The new functionality in the updated SDK includes get/set oscillator frequency, and standby/wakeup functions as well as short-range accuracy

**Comment by PaulZC** [comment]
Let's keep this open until #7 is merged...
